src/neuroconv/datainterfaces/behavior/medpc/medpc_helpers.py

```diff
@@ -0,0 +1,173 @@
+        A dictionary with the variable names as keys and the data extracted from medpc output are the values.
+
+    Raises
+    ------
+    ValueError
+        If the session with the given conditions could not be found.
+    """
+    with open(file_path, "r") as f:
+        lines = f.readlines()
+    session_lines = get_session_lines(lines, session_conditions=session_conditions, start_variable=start_variable)
+
+    # Parse the session lines into a dictionary
+    session_dict = {}
+    for i, line in enumerate(session_lines):
+        line = line.rstrip()
+        if line.startswith("\\"):  # \\ indicates a commented line in the MedPC file
+            continue
+        assert ":" in line, f"Could not find ':' in line {repr(line)}"
+        split_line = line.split(":", maxsplit=1)
+        medpc_name, data = split_line
+        data = data.strip()
+        if "\t" in data:  # some sessions have a bunch of garbage after the last datum in the line separated by tabs
+            data = data.split("\t")[0]
+        if line.find(":") == 6:  # multiline variable
+            if medpc_name == "     0":  # first line of multiline variable
+                multiline_variable_name = session_lines[i - 1].split(":")[0]
+                if multiline_variable_name in medpc_name_to_info_dict:
+                    output_name = medpc_name_to_info_dict[multiline_variable_name]["name"]
+                    session_dict[output_name] = []
+            if multiline_variable_name not in medpc_name_to_info_dict:
+                continue
+            data = data.split(" ")
+            for datum in data:
+                datum = datum.strip()
+                if datum == "":
+                    continue
+                output_name = medpc_name_to_info_dict[multiline_variable_name]["name"]
+                session_dict[output_name].append(datum)
+
+        # single line variable
+        elif medpc_name in medpc_name_to_info_dict:
+            output_name = medpc_name_to_info_dict[medpc_name]["name"]
+            session_dict[output_name] = data
+
+    # Convert the data types
+    for info in medpc_name_to_info_dict.values():
+        output_name = info["name"]
+        is_array = info["is_array"]
+        if output_name in session_dict:
+            if is_array:
+                if session_dict[output_name] == "":
+                    session_dict[output_name] = np.array([], dtype=float)
+                elif type(session_dict[output_name]) == str:  # not a multiline variable
+                    raise ValueError(
+                        f"Expected {output_name} to be a multiline variable, but found a single line variable."
+                    )
+                else:
+                    session_dict[output_name] = np.array(session_dict[output_name], dtype=float)
+                    session_dict[output_name] = np.trim_zeros(
+                        session_dict[output_name], trim="b"
+                    )  # MEDPC adds extra zeros to the end of the array
```

Comment:
Especially with all this parsing; a generic package for reading MedPC files would be something to support all that, but that's way too much for us to claim to support - instead, just a collection of internal helper functions for getting the data into an NWB file in a particular way

Reply:
yeah I think a separate package for medpc is outside of our scope.